tensorflow/c/c_api_experimental.cc

```diff
@@ -619,8 +620,14 @@ TF_Tensor* TF_CheckpointReaderGetTensor(TF_CheckpointReader* reader, const char*
   return tensorflow::TF_TensorFromTensor(*tensor.get(), status);
 }
 
-void TF_CheckpointReaderGetVariableShape(TF_CheckpointReader* reader, const char* name, int64_t* dims, int num_dims) {
+void TF_CheckpointReaderGetVariableShape(TF_CheckpointReader* reader, const char* name, int64_t* dims, int num_dims, TF_Status* status) {
   const auto& m = reader->GetVariableToShapeMap();
+  int rank = m.at(name).dims();
```

Comment:
Instead of keeping a reference to `m`, can you keep a reference to `m.at(name)` so we don't look this up a bunch of times needlessly?

Reply:
done